### ISSUE TITLE
AuthenticationCallback methods are only called the first time I login

### ISSUE THREAD

**Issue by DaireJN** [auth0/Lock.Android#607]
### Describe the problem

the first time i login callbacks are recieved as expected callbacks are not recieved in subsequent logins, if i close the app and attempt to login again the same occurs (first time = callbacks recieved, nothing subsequent times) The logs in the auth0 dashboard indicate everything is ok

### What was the expected behavior?

callback recieved everytime i login or cancel/error

### Sample

Here is a project demontrating the issue, to build it replace "com_auth0_domain" and  "com_auth0_client_id" in strings.xml with valid credentials and run.

https://github.com/DaireJN/LockExample.git

## Logs

these logs are recieved on fist sign in, no logs occur for subsequent sign ins until app is killed + reopened

2021-02-15 17:42:14.646 26962-26962/packagename V/Lock: AUTHENTICATION action received in our BroadcastReceiver
2021-02-15 17:42:14.646 26962-26962/packagename D/AuthenticationCallback: User authenticated!

https://user-images.githubusercontent.com/28805987/107978323-799c1280-6fb4-11eb-9fac-6a28c558036d.mov

### Environment

- **Version of this library used:** 2.24.0
- **Which framework are you using, if applicable:** 
- **Other modules/plugins/libraries that might be involved:** Kotlin
- **Any other relevant information you think would be useful:** min sdk 21 target sdk 29

**Comment by DaireJN** [comment]
Any insights on this?

**Comment by lbalmaceda** [comment]
@DaireJN  Sorry for the delayed response. Thanks for providing a repro project, I was able to reproduce this. 

It seems this [offending line](https://github.com/auth0/Lock.Android/blob/master/lib/src/main/java/com/auth0/android/lock/Lock.java#L150) in the library code is unregistering the even listener after an event was received. As long as devs are calling `lock.onDestroy()` to clear that broadcast listener registration when they decide, we should be able to remove that line safely. 

So TLDR, we have to:
- Remove the line I linked above in both `PasswordlessLock` and `Lock` files.
- Improve the docs around calling `onDestroy` to ensure a proper lifecycle callback handling.

In the meantime, and since the listener is registered every time you `build` a lock instance from the Builder, you can use that to reset the callback registration.

**Comment by lbalmaceda** [comment]
@DaireJN I realize the [readme](https://github.com/auth0/Lock.Android) today is not clear enough about the lock instance disposal in the `Activity#onDestroy` method. Considering that this library has been out for more than 4 years like this and this is the first time someone reports this bug, I'll make the fix available only for the next major to prevent introducing a different behavior for major v2 customers.

We're working on the next major and I opened a PR to fix this there (linked above). I'd appreciate your feedback on whether the new guidance is more clear or not. See [migration guide](https://github.com/auth0/Lock.Android/pull/619/files#diff-45651f4c3460b0f20be048556615842a07ec86d262fe5d3d346b5e4a9ce56d6aR74) and [readme](https://github.com/auth0/Lock.Android/pull/617/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R209) changes.
